internal/storage/storage.go

```diff
@@ -233,6 +233,21 @@ func (s *Storage) MustGetPickleStepResultsByPickleID(pickleID string) (psrs []mo
 	return psrs
 }
 
+// MustGetPickleStepResultsByPickleIDUntilStep will retrieve pickle step results by pickle id
+// from 0..stepID for that pickle.
+func (s *Storage) MustGetPickleStepResultsByPickleIDUntilStep(pickleID string, untilStepID string) (psrs []models.PickleStepResult) {
```

Comment:
This only needed to be done for the Pretty formatter, so I wasn't sure if I should just do it after `MustGetPickleStepResultsByPickleID` in `fmt_pretty.go`, or create a function for it.

I've opted to create a function with a pretty bad name, so let me know if yous are happy with this.